src/overlays/actors/ovl_Bg_Jya_Megami/z_bg_jya_megami.c

```diff
@@ -21,31 +29,325 @@ const ActorInit Bg_Jya_Megami_InitVars = {
+    Graph_OpenDisps(dispRefs, globalContext->state.gfxCtx, "../z_bg_jya_megami.c", 706);
+    func_80093D18(globalContext->state.gfxCtx);
+    gSPSegment(gfxCtx->polyOpa.p++, 0x08, SEGMENTED_TO_VIRTUAL(sRightSideCrumbles[this->crumbleIndex]));
+    gSPSegment(gfxCtx->polyOpa.p++, 0x09, SEGMENTED_TO_VIRTUAL(sLeftSideCrumbles[this->crumbleIndex]));
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalContext->state.gfxCtx, "../z_bg_jya_megami.c", 716),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(gfxCtx->polyOpa.p++, &D_06005780);
+    Graph_CloseDisps(dispRefs, globalContext->state.gfxCtx, "../z_bg_jya_megami.c", 720);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Megami/func_8089A604.s")
+void BgJyaMegami_DrawExplosion(BgJyaMegami* this, GlobalContext* globalContext) {
+    GraphicsContext* gfxCtx;
+    BgJyaMegami_Piece* temp;
+    u32 i;
+    Gfx* dispRefs[5];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Megami/func_8089A78C.s")
+    gfxCtx = globalContext->state.gfxCtx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Megami/func_8089A820.s")
+    Graph_OpenDisps(dispRefs, globalContext->state.gfxCtx, "../z_bg_jya_megami.c", 746);
+    func_80093D18(globalContext->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Megami/BgJyaMegami_Update.s")
+    for (i = 0; i < ARRAY_COUNT(sPiecesInit); ++i) {
+        temp = &this->pieces[i];
+        Matrix_Translate(temp->pos.x + sPiecesInit[i].unk_00.x, temp->pos.y + sPiecesInit[i].unk_00.y,
+                         temp->pos.z + sPiecesInit[i].unk_00.z, MTXMODE_NEW);
+        Matrix_RotateY(temp->rotVelY * 0.0000958738f, MTXMODE_APPLY);
+        Matrix_RotateX(temp->rotVelX * 0.0000958738f, MTXMODE_APPLY);
+        Matrix_Scale(0.1f, 0.1f, 0.1f, MTXMODE_APPLY);
+        Matrix_Translate(sPiecesInit[i].unk_00.x * -10.0f, sPiecesInit[i].unk_00.y * -10.0f,
+                         sPiecesInit[i].unk_00.z * -10.0f, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Megami/func_8089ABC8.s")
+        gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalContext->state.gfxCtx, "../z_bg_jya_megami.c", 778),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(gfxCtx->polyOpa.p++, sDisplayLists[i]);
+    }
+    Graph_CloseDisps(dispRefs, globalContext->state.gfxCtx, "../z_bg_jya_megami.c", 783);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Megami/func_8089AD48.s")
+void BgJyaMegami_Draw(Actor* thisx, GlobalContext* globalContext) {
+    BgJyaMegami* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Megami/BgJyaMegami_Draw.s")
+    func_800628A4(0, &this->collider);
+    if (this->actionFunc == BgJyaMegami_Explosion) {
+        BgJyaMegami_DrawExplosion(this, globalContext);
+    } else {
+        BgJyaMegami_DrawDetectLight(this, globalContext);
```

Comment:
confused by this name, what is it actually drawing? the mirror shield patter on the face?
Think it can probably be phrased better

Reply:
there are two different actionFuncs and the main draw function goes into a sub draw function for each of them, so the naming scheme I used is BgJyaMegami_Draw[ActionFuncName]. I think for DrawExplosion this works well but maybe DrawDetectLight should just become DrawFace or DrawNormal? not sure

Reply:
DrawFace works best if its drawing the face that detects the light imo